testplan/testing/multitest/result.py

```diff
@@ -1935,3 +1943,13 @@ def __iter__(self):
 
     def __len__(self):
         return len(self.entries)
+
+
+class Attachment(object):
+    """Represents a file attached to the result object."""
+
+    def __init__(self, filepath):
+        self.source_path = filepath
+        self.uuid = str(uuid.uuid4())
+        self.filename = os.path.basename(filepath)
+        self.dst_path = os.path.join(self.uuid, self.filename)
```

Comment:
why make the file path uuid as opposed to a some standard /attachments/tmp/FILENAME

Reply:
I used a UUID to avoid clashes, different testcases might want to attach files with the same name, which would cause problems if we only used the filename as the key.

There are other possibilities, maybe I could have created a directory hierarchy that matches the testcase hierarchy (e.g. attachments/MultiTestName/TestSuiteName/TestCaseName/filename.txt), I thought that would be more complicated though. Maybe it would be better to hash the file to generate a UID instead of using uuid, then attaching the same file multiple times could be done more efficiently.

Reply:
does hashing files that have the same problem as using a static file_path, different files with the same name would then clash?

Reply:
I mean hashing the contents of the files, not the names. Though maybe that would be too slow for very large attachments...

Reply:
Now that I think about it, I'm not sure it's worth it. There could still (potentially) be hash collisions for different files, UUID is guaranteed to be unique. I don't think it would be a common requirement to attach the same file from multiple testcases (if it ever is, we can revisit).